### PR TITLE
fix: add role alert to the search results dynamically

### PR DESCRIPTION
**Pull request by chosww**
* [x] This pull request has been tested by running `npm run test` without errors
* [x] This pull request has been built by running `npm run build` without errors
* [x] This isn't a duplicate of an existing pull request

## Description

The search results in the resources page wasn't announced by screen readers because the content of the page gets rendered from reloading the page. Fix the issue by adding the role alert to the search results dynamically.

## Steps to test

1. Navigate to the resources page
2. Use the search filters and press Apply filter button with a screen reader on

**Expected behavior:** 

Screen reader should announce the search results.

## Additional information

Not all screen reader & browser combinations will work well, for example VoiceOver and Firefox wouldn't work as expected.

### CODE REVIEW

File: src/admin/previews.js [hunk 1]
```diff
@@ -10,6 +10,7 @@ import getResourceMetadataLabelFilter from "../_filters/getResourceMetadataLabel
 import site from "../_data/site.json";
 import imagePositionWithTextShortcode from "../_shortcodes/image-position-with-text.js";
 import getId from "../_utils/extract-youtube-id.js";
+import slugify from "../../node_modules/@11ty/eleventy/node_modules/@sindresorhus/slugify";
```
Comment (greatislander) [inline src/admin/previews.js:13]: I think you should import this package as just `@sindresorhus/slugify`, rather than the one within Eleventy

Comment (chosww) [reply src/admin/previews.js:13]: Thanks Ned, we imported slugify this way in another project if I remember, what is the correct syntax for it?

Comment (chosww) [reply src/admin/previews.js]: Never mind, happened to be that my node_modules was outdated and didn't have @sindresorhus/slugify

File: src/assets/scripts/_resources-utils.js [hunk 2]
```diff
@@ -225,9 +225,9 @@ export function renderSearchResults(numberOfResources, resourceTopics, resourceT
     let appliedFilterHtml = "<h2>Search results</h2><div class='resources-applied-filters'>";
 
     if (numberOfResources === 0) {
-        appliedFilterHtml += "<div class='resources-no-results' role='alert'><p>Sorry, no results were found based on your applied filters.</p></div>";
+        appliedFilterHtml += "<div class='resources-no-results' id='resources-sort-result'><p>Sorry, no results were found based on your applied filters.</p></div>";
```
Comment (greatislander) [inline src/assets/scripts/_resources-utils.js:228]: I would change this to `resources-result` as it's relevant for both sorting and filtering.

**Review by greatislander — CHANGES_REQUESTED** [review]

**Comment by chosww** [comment]
@greatislander with current structure and implementation for the filters, it is hard to have corresponding fieldset and legend for the filters. Instead, I provided more details about each filter by providing details via aria-label. Please let me know what you think about this approach.

File: src/_includes/partials/components/filter-resources.njk [hunk 3]
```diff
@@ -11,7 +11,7 @@
             {% for topic in resourceTopics.resourceTopics %}
                 <li>
                     <input class="filter-checkbox" type="checkbox" data-filter="topics" id="{{ topic.value }}" name="{{ 'to_' + topic.value }}">
-                    <label for="{{ topic.value }}">{{ topic.label }}</label>
+                    <label aria-label="Topics filter, {{ topic.label }}" for="{{ topic.value }}">{{ topic.label }}</label>
```
Comment (greatislander) [inline src/_includes/partials/components/filter-resources.njk:14]: I'd suggest removing these, and instead enclosing each `<ul>` in a `<fieldset>` with a visually hidden `<legend>`:

```njk
<fieldset>
        <legend class="visually-hidden">{{ filterTitle }}</legend>
	<!-- <ul> goes here -->
</fieldset>
```

**Review by greatislander — CHANGES_REQUESTED** [review]

**Review by greatislander — APPROVED** [review]

**Pull request merged**